ext/drcovlib/modules.c

```diff
@@ -174,11 +176,12 @@ event_module_load(void *drcontext, const module_data_t *data, bool loaded)
                 }
             }
 #endif
-            break;
+            /* We may just see a sub module, so we cannot stop here. We need iterate
+             * all entries to update all sub modules of the reloaded module.
```

Comment:
The bug actually happens like below:
1. a module with multiple sub-modules is loaded, the module self and all sub-modules are added into module_table.
2. the module is unloaded, the module and all its sub-modules are marked as unloaded
3. the module is loaded at the same place (where the bug happens)
    the first entry we found match is actually a sub-module, and there are more sub-modules located before this submodule need update, but the code only updated all module/sub-modules after.
We could fixed the bug by forward scan in the outer loop, so we will see the module first before any submodule.
I will see if there is a better fix so we can reuse the code of handling the non-contiguous module reload.